store/bolt/internal/user.go

```diff
@@ -0,0 +1,40 @@
+package internal
+
+import (
+	"github.com/gogo/protobuf/proto"
+	"github.com/gopheracademy/congo"
+)
+
+// MarshalUser encodes u into a binary form.
+func MarshalUser(u *congo.User) ([]byte, error) {
+	return proto.Marshal(encodeUser(u))
+}
+
+// UnmarshalUser decodes data into a user.
+func UnmarshalUser(data []byte) (*congo.User, error) {
+	var u User
+	if err := proto.Unmarshal(data, &u); err != nil {
+		return nil, err
+	}
+	return decodeUser(&u), nil
+}
```

Comment:
The `MarshalUser()` and `UnmarshalUser()` functions are the only exported functions. They provide a way for the storage layer to encode and decode from raw bytes to our `User` application type. The `store/bolt` package will handle where these bytes are actually stored in our database.